src/daft-micropartition/src/micropartition.rs

```diff
@@ -95,11 +95,14 @@ fn materialize_scan_task(
     scan_task: Arc<ScanTask>,
     io_stats: Option<IOStatsRef>,
 ) -> crate::Result<(Vec<Table>, SchemaRef)> {
-    let column_names = scan_task
+    let pushdown_columns = scan_task
         .pushdowns
         .columns
         .as_ref()
-        .map(|v| v.iter().map(|s| s.as_ref()).collect::<Vec<&str>>());
+        .map(|v| v.iter().map(|s| s.as_str()).collect::<Vec<&str>>());
+    let file_column_names =
```

Comment:
you likely need to apply this logic to the eager code path. Say you passed in a predicate.

Reply:
This should be fixed in this section: https://github.com/Eventual-Inc/Daft/pull/2118/files#diff-1d03b58906d515cc349a8a414595727e787d60af7adb3ab9fb3db3558c9e7e27R789-R808

Unfortunately our code is a little jank right now, with 2 main codepaths:

1. `MicroPartition::from_scan_task` either eagerly produces loaded micropartitions or lazily produces unloaded micropartitions using `read_parquet_into_micropartition`
2. Lazy micropartitions then load data using `materialize_scan_task`

So any logic often has to be duplicated across those two functions (`read_parquet_into_micropartition` and `materialize_scan_task`). Perhaps there is some opportunity to consolidate this by having `read_parquet_into_micropartition` call `materialize_scan_task` under the hood, but that might be a refactor for another PR?

Reply:
I think a simpler way would be to just wrap `read_parquet_bulk` in this crate

Reply:
Could you elaborate? Both `read_parquet_into_micropartition` and `materialize_scan_task` do currently wrap `read_parquet_bulk`.

Reply:
We do already have a shared `_read_parquet_into_loaded_micropartition` utility for `read_parquet_into_micropartition`, maybe we should refactor `materialize_scan_task` to call out to similar helpers for each file format.

Reply:
> I think that @samster25 was talking about creating a `read_parquet_bulk_with_virtual` (or similarly named) utility in this `micropartition.rs` module that does all of the virtual column + pushdown handling before calling `read_parquet_bulk`, shared (called) by both the eager and lazy paths, so we don't have to duplicate this handling logic.

Yeah exactly!